views.go

```diff
@@ -63,7 +63,7 @@ func LoginGET(c echo.Context) error {
 
 	if username.String() == "" && verification_key.String() == "" {
 		data.LoginForm = true
-		data.CASLogin = GetCasLoginUrl(service.String())
+		data.CASLogin = GetCasLoginUrl(c.Request(), service.String())
```

Comment:
I'm not entirely sure what the right answer to your question is :)

However, `GetCasLoginUrl` cannot rely on the value of the `Host` command line argument when generating the login form.  (Reasons why are documented in #6: Using `0.0.0.0` for the `-host` command line parameter generates an HTML form that submits to `0.0.0.0`, which won't work).

This PR is arguing for a more robust way to determine the IP address or domain name used in the CAS login URL, and is proposing using the value of the `Host:` HTTP header.  The PR passes the entire HTTP request object to `GetCasLoginUrl`, but a `string` with the host name would also work.